core/src/main/java/org/apache/iceberg/BaseMetastoreCatalog.java

```diff
@@ -299,6 +309,14 @@ protected MetricsReporter metricsReporter() {
     return metricsReporter;
   }
 
+  // Atomically set given metadata as current
+  protected void setAsCurrent(
```

Comment:
I'm not sure it's super clear how this works from the signature. IMHO, an implementation of this shouldn't care what "base" is. It should either be "setAsCurrent if empty or setAsCurrentRegardless"

In my above snippet maybe something like

if (overwrite)
  setAsCurrent(Identifier, Location)
 } else {
   setAsCurrentIfEmpty(identifier, location)
 }

Reply:
Removed `base` from `setAsCurrent`. Originally I added this to use for lock-free version of AlterTable in HiveCatalog as base is not used in JDBC or InMemory catalog at all. I think it's better to keep the interface clean and general to all catalogs instead of considering implementation as the constraint